tests/linearizability/model.go

```diff
@@ -17,33 +17,64 @@ package linearizability
 import (
 	"encoding/json"
 	"fmt"
+	"strings"
 
 	"github.com/anishathalye/porcupine"
 )
 
-type Operation string
+type OperationType string
 
 const (
-	Get    Operation = "get"
-	Put    Operation = "put"
-	Delete Operation = "delete"
-	Txn    Operation = "txn"
+	Get    OperationType = "get"
+	Put    OperationType = "put"
+	Delete OperationType = "delete"
+	Txn    OperationType = "txn"
 )
 
 type EtcdRequest struct {
-	Op            Operation
+	Conds []EtcdCondition
+	Ops   []EtcdOperation
+}
+
+type EtcdCondition struct {
 	Key           string
-	PutData       string
-	TxnExpectData string
-	TxnNewData    string
+	ExpectedValue string
+}
+
+type EtcdOperation struct {
+	Type  OperationType
+	Key   string
+	Value string
 }
 
 type EtcdResponse struct {
-	GetData      string
-	Revision     int64
-	Deleted      int64
-	TxnSucceeded bool
-	Err          error
+	Err        error
+	Revision   int64
+	TxnFailure bool
+	Result     []EtcdOperationResult
+}
+
+type EtcdOperationResult struct {
+	Value   string
+	Deleted int64
+}
+
+func Match(r1, r2 EtcdResponse) bool {
```

Comment:
Why `Match` and not `Equals` ?

Reply:
Can the function be as simple as 
```
return reflect.DeepEqual(r1, r2)
```

Reply:
Done. I plan to add unknown result in future but let's keep it simple for now.